### PR TITLE
Try to fill in the qClassify3DMASCDialog if names are available in the parameter file and clouds exist in the database tree with the same name (case insensitive)

### PR DESCRIPTION
**Pull request by p-leroy**

### CODE REVIEW

File: qClassify3DMASCDialog.cpp [hunk 1]
```diff
@@ -140,70 +139,109 @@ void Classify3DMASCDialog::writeSettings()
 	settings.setValue("keepAttributes", keepAttributesCheckBox->isChecked());
 }
 
-void Classify3DMASCDialog::setCloudRoles(const QList<QString>& roles, QString corePointsLabel)
+void Classify3DMASCDialog::setComboBoxIndex(QMap<QString, QString> rolesAndNames, QLabel* label, QMap<QString, QVariant> namesAndUniqueIds, QComboBox* comboBox)
```
Comment (dgirardeau) [inline qClassify3DMASCDialog.cpp:142]:
```suggestion
void Classify3DMASCDialog::setComboBoxIndex(const QMap<QString, QString>& rolesAndNames, QLabel* label, const QMap<QString, QVariant>& namesAndUniqueIds, QComboBox* comboBox)
```

Comment (dgirardeau) [reply qClassify3DMASCDialog.cpp:142]: otherwise the maps will be copied each time!

File: qClassify3DMASCDialog.cpp [hunk 2]
```diff
@@ -140,70 +139,109 @@ void Classify3DMASCDialog::writeSettings()
...
+{
+	QString name;
```
Comment (dgirardeau) [inline qClassify3DMASCDialog.cpp:144]:
```suggestion
	QString name = rolesAndNames[label == testLabel ? QString("TEST") :  label->text()); // testLabel is specific, the role is always TEST
```

Comment (dgirardeau) [reply qClassify3DMASCDialog.cpp:144]: (and remove the lines below)

File: qClassify3DMASCDialog.cpp [hunk 3]
```diff
@@ -140,70 +139,109 @@ void Classify3DMASCDialog::writeSettings()
...
+
+	// build a map 'name : uniqueId' of the available clouds in the database tree (duplicate names are not handled, simply keep the first occurrence)
+	ccHObject::Container clouds;
+	if (m_app->dbRootObject())
+	{
+		m_app->dbRootObject()->filterChildren(clouds, true, CC_TYPES::POINT_CLOUD);
+	}
+	for (size_t i = 0; i < clouds.size(); ++i)
+	{
+		if (clouds[i]->isA(CC_TYPES::POINT_CLOUD)) //as filterChildren only test 'isKindOf'
```
Comment (dgirardeau) [inline qClassify3DMASCDialog.cpp:232]:
```suggestion
		if (clouds[i]->isA(CC_TYPES::POINT_CLOUD)) //as filterChildren only tests 'isKindOf'
```

Comment (dgirardeau) [reply qClassify3DMASCDialog.cpp:232]: You can set the 4th argument of filterChildren to true (= strict)

File: qClassify3DMASCDialog.cpp (hunk 1 above)
Comment (p-leroy) [reply qClassify3DMASCDialog.cpp:142]: Obviously, sorry

**Pull request merged**